hack/builders/build_release_image.sh

```diff
@@ -0,0 +1,163 @@
+    echo "--openstack-ccm custom openstack cloud-controller-manager image name, default: quay.io/openshift/origin-openstack-cloud-controller-manager:$RELEASE"
+    echo "--kapio         custom kube-apiserver-operator image name, default: current kube-apiserver-operator image from the release payload"
+    echo "--kcmo          custom kube-controller-manager-operator image name, default: current kube-controller-manager-operator image from the release payload"
+    echo "--mco           custom machine-config-operator image name, default: current machine-config-operator image from the release payload"
+}
+
+: ${GOPATH:=${HOME}/go}
+: ${TAG:="latest"}
+: ${RELEASE:="4.9"}
+: ${OC_REGISTRY_AUTH_FILE:="pull-secret.txt"}
+: ${CCCMO_IMAGE:="quay.io/openshift/origin-cluster-cloud-controller-manager-operator:$RELEASE"}
+: ${AWSCCM_IMAGE:="quay.io/openshift/origin-aws-cloud-controller-manager:$RELEASE"}
+: ${AZURECCM_IMAGE:="quay.io/openshift/origin-azure-cloud-controller-manager:$RELEASE"}
+: ${AZURENODE_IMAGE:="quay.io/openshift/origin-azure-cloud-node-manager:$RELEASE"}
+: ${OCCM_IMAGE:="quay.io/openshift/origin-openstack-cloud-controller-manager:$RELEASE"}
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h|--help)
+            help
+            exit 0
+            ;;
+            
+        -u|--username)
+            USERNAME=$2
+            shift 2
+            ;;
+
+        -t|--tag)
+            TAG=$2
+            shift 2
+            ;;
+
+        -r|--release)
+            RELEASE=$2
+            shift 2
+            ;;
+
+        -a|--auth)
+            OC_REGISTRY_AUTH_FILE=$2
+            shift 2
+            ;;
+
+        --cccmo)
+            CCCMO_IMAGE=$2
+            shift 2
+            ;;
+
+        --aws-ccm)
+            AWSCCM_IMAGE=$2
+            shift 2
+            ;;
+
+        --azure-ccm)
+            AZURECCM_IMAGE=$2
+            shift 2
+            ;;
+
+        --azure-node)
+            AZURENODE_IMAGE=$2
+            shift 2
+            ;;
+
+        --openstack-ccm)
+            OCCM_IMAGE=$2
+            shift 2
+            ;;
+
+        --kapio)
+            KAPIO_IMAGE=$2
+            shift 2
+            ;;
+
+        --kcmo)
+            KCMO_IMAGE=$2
+            shift 2
+            ;;
+
+        --mco)
+            MCO_IMAGE=$2
+            shift 2
+            ;;
+
+        *)
+            echo "Invalid option $1"
+            help
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$USERNAME" ]; then
+    echo "-u/--username was not provided, exiting ..."
+    exit 1
+fi
+
+if [ ! -f "$OC_REGISTRY_AUTH_FILE" ]; then
+    echo "$OC_REGISTRY_AUTH_FILE not found, exiting ..."
+    exit 1
+fi
+
+echo "Creating local image registry at localhost:5000"
+docker rm -fi registry
+docker run -d -p 5000:5000 --restart=always --name registry docker.io/library/registry:2
+
+PREFIX="Pull From: "
+DEST_IMAGE="quay.io/$USERNAME/origin-release:$TAG"
+TEMP_IMAGE="localhost:5000/origin-release:$TAG"
```

Comment:
Can we also avoid using temp image here, so the image will immediately go into quay or other registry? This adds complexity. Would be good to have a flag to skip local registry creation.